### PR TITLE
fix: wrong overall coverage report dependencies

### PR DESCRIPTION
**Pull request by cuinixam**

### CODE REVIEW

**Comment by cuinixam** [comment]
We are using `ninja 1.11.1` where https://github.com/ninja-build/ninja/pull/1964 should be included. I do not understand why this problem still occurs.

File: cmake/common.cmake [hunk 1]
```diff
@@ -252,7 +252,8 @@ function(_spl_coverage_create_overall_report)
         add_custom_command(
             OUTPUT ${COV_OUT_VARIANT_HTML}
             COMMAND gcovr --root ${CMAKE_SOURCE_DIR} --add-tracefile \"${CMAKE_CURRENT_BINARY_DIR}/**/coverage.json\" --html --html-details --output ${COV_OUT_VARIANT_HTML}
-            DEPENDS coverage
+            DEPENDS ${GLOBAL_COMPONENTS_COVERAGE_JSON_LIST}
+            COMMENT "Generating coverage overall report ${OUTPUT}..."
```
Comment (xxthunder) [inline cmake/common.cmake]:
```suggestion
            COMMENT "Generating overall code coverage report ${OUTPUT} ..."
```

Really? ${OUTPUT} is available here?

File: cmake/common.cmake [hunk 2]
```diff
@@ -318,6 +319,8 @@ macro(_spl_add_test_suite PROD_SRC TEST_SOURCES)
     )
 
     # Create coverage report
+    set(GLOBAL_COMPONENTS_COVERAGE_JSON_LIST "${GLOBAL_COMPONENTS_COVERAGE_JSON_LIST};${CMAKE_CURRENT_BINARY_DIR}/coverage.json" CACHE INTERNAL "List of all coverage.json files")
+
     set(COV_OUT_JSON coverage.json)
```
Comment (xxthunder) [inline cmake/common.cmake]: We should use this variable name everywhere instead of "coverage.json".

File: cmake/common.cmake [hunk 3]
```diff
@@ -326,13 +329,15 @@ macro(_spl_add_test_suite PROD_SRC TEST_SOURCES)
         # Run gcovr to generate coverage json for the component
         COMMAND gcovr --root ${CMAKE_SOURCE_DIR} --json --output ${COV_OUT_JSON} ${GCOVR_ADDITIONAL_OPTIONS} ${CMAKE_CURRENT_BINARY_DIR}
         DEPENDS ${TEST_OUT_JUNIT}
+        COMMENT "Generating component ${component_name} coverage json report ${COV_OUT_JSON}..."
```
Comment (xxthunder) [inline cmake/common.cmake]:
```suggestion
        COMMENT "Generating component ${component_name} code coverage json report ${COV_OUT_JSON} ..."
```

File: cmake/common.cmake [hunk 4]
```diff
@@ -326,13 +329,15 @@ macro(_spl_add_test_suite PROD_SRC TEST_SOURCES)
...
         DEPENDS ${TEST_OUT_JUNIT}
+        COMMENT "Generating component ${component_name} coverage json report ${COV_OUT_JSON}..."
     )
 
     set(COV_OUT_HTML reports/coverage/index.html)
     add_custom_command(
         OUTPUT ${COV_OUT_HTML}
         COMMAND gcovr --root ${CMAKE_SOURCE_DIR} --add-tracefile ${COV_OUT_JSON} --html --html-details --output ${COV_OUT_HTML} ${GCOVR_ADDITIONAL_OPTIONS}
         DEPENDS ${COV_OUT_JSON}
+        COMMENT "Generating component ${component_name} coverage html report ${COV_OUT_HTML}..."
```
Comment (xxthunder) [inline cmake/common.cmake]:
```suggestion
        COMMENT "Generating component ${component_name} code coverage html report ${COV_OUT_HTML} ..."
```

**Pull request merged**